src/main/java/retriever/Ui.java

```diff
@@ -0,0 +1,205 @@
+package retriever;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import retriever.task.Task;
+
+/**
+ * The class handles the user interaction between the
+ * user and the chatbot.
+ */
+public class Ui {
+    private static String retrieverResponse;
+
+    // Predetermined strings.
+    private String dashedLine = "_______________________________________________";
+    private String welcomeMessage = "Hello, I am Retriever\nHow Can I Help You Today?";
+    private String goodByeMessage = "-> Sad To See You Go!";
+    private String emptyListMessage = "My Memory Is Empty, Please Feed Items!";
+    private String printListMessage = "-> Your Tasks, My Master:";
+    private String goodBoyMessage = "Woof! Whose a Good Boy?";
+    private String badBoyMessage = "Woof! Whose a Bad Boy?";
+    private String logo = "  __      ^\n"
+            + "o'')}____//\n"
+            + " `_'      )\n"
+            + "(_(_/-(_/\n";
+
+    private Scanner scanner;
+
+    /**
+     * Sets up the scanner to take in user input.
+     */
+    public Ui() {
+        scanner = new Scanner(System.in);
+    }
+
+    /**
+     * Prints the message received as an argument.
+     *
+     * @param message The message user would like to print.
+     */
+    public void printMessage(String message) {
+        retrieverResponse = message;
+        System.out.println(message);
+    }
+
+    /**
+     * Prints the welcome message.
+     */
+    public void printWelcomeMessage() {
+        System.out.println(dashedLine);
+        System.out.println(welcomeMessage);
+        System.out.println(logo);
+        System.out.println(dashedLine);
+
+        retrieverResponse = welcomeMessage;
+    }
+
+    /**
+     * Prints the goodbye message.
+     */
+    public void printGoodByeMessage() {
+        // Closing the scanner.
+        scanner.close();
+
+        System.out.println(goodByeMessage);
+        System.out.println(dashedLine);
+
+        retrieverResponse = goodBoyMessage;
+    }
+
+    /**
+     * Prints the error message.
+     *
+     * @param errorMessage The exception message to be printed.
+     */
+    public void printErrorMessage(String errorMessage) {
+        System.out.println(errorMessage);
+        retrieverResponse = errorMessage;
+    }
+
+    /**
+     * Prints a dashed line.
+     */
+    public void printDashedLine() {
+        System.out.println(dashedLine);
+    }
+
+    /**
+     * Prints the task added message in a specified format.
+     *
+     * @param task The task that has been added.
+     * @param taskListLength The length of the task list, after adding the task.
+     */
+    public void printTaskAdded(Task task, int taskListLength) {
+        System.out.println("-> Where's My Treat? I Added:\n\t" + task);
+        System.out.println("\nYou Owe Me " + taskListLength + " Treat(s), Master!");
```

Comment:
Would it be better if you assign the string to retrieverResponse first, then System.out.println(retrieverResponse)?
```suggestion
        retrieverResponse = "Where's My Treat? I Added:\n"
                + task + "\nYou Owe Me " + taskListLength + " Treat(s), Master!";
        System.out.println(retrieverResponse");
```